packages/core/src/awsService/cloudWatchLogs/registry/liveTailSession.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { CloudWatchLogsClient, StartLiveTailCommand, StartLiveTailCommandOutput } from '@aws-sdk/client-cloudwatch-logs'
+import { LogStreamFilterResponse } from '../liveTailLogStreamSubmenu'
+import { CloudWatchLogsSettings } from '../cloudWatchLogsUtils'
+import { Settings } from '../../../shared'
+import { createLiveTailURIFromArgs } from './liveTailSessionRegistry'
+
+export type LiveTailSessionConfiguration = {
+    logGroupName: string
+    logStreamFilter?: LogStreamFilterResponse
+    logEventFilterPattern?: string
+    region: string
+}
+
+export type LiveTailSessionClient = {
+    cwlClient: CloudWatchLogsClient
+    abortController: AbortController
+}
+
+export class LiveTailSession {
+    private liveTailClient: LiveTailSessionClient
+    private _logGroupName: string
+    private logStreamFilter?: LogStreamFilterResponse
+    private logEventFilterPattern?: string
+    private _maxLines: number
+    private _uri: vscode.Uri
+
+    static settings = new CloudWatchLogsSettings(Settings.instance)
+
+    public constructor(configuration: LiveTailSessionConfiguration) {
+        this._logGroupName = configuration.logGroupName
+        this.logStreamFilter = configuration.logStreamFilter
+        this.liveTailClient = {
+            cwlClient: new CloudWatchLogsClient({ region: configuration.region }),
+            abortController: new AbortController(),
+        }
+        this._maxLines = LiveTailSession.settings.get('liveTailMaxEvents', 10000)
+        this._uri = createLiveTailURIFromArgs(configuration)
+    }
+
+    public get maxLines() {
+        return this._maxLines
+    }
+
+    public get uri() {
+        return this._uri
+    }
+
+    public get logGroupName() {
+        return this._logGroupName
+    }
+
+    public startLiveTailSession(): Promise<StartLiveTailCommandOutput> {
```

Comment:
what calls these functions? Or is this what calls it coming in another PR?

Reply:
Will be coming in another PR. The TailLogGroup command will call these functions.